lib/puppet/provider/common.rb

```diff
@@ -25,15 +25,15 @@ def pretty(arg)
 end
 
 # Removes quotes from nil and false values
-def data_parse(data = {})
+def data_parse(data = {}, client = nil)
```

Comment:
Also, I'm not sure about a method named `data_parse` doing rest calls.
Maybe we could find a better name.
After some time debugging the code, it was surprising for me.